clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -3188,8 +3194,21 @@ private int processKafkaDataMessage(
     }
 
     if (emitMetrics.get()) {
+      /**
+       * Metrics that are only emitted for the largest active version of the store.
+       */
       hostLevelIngestionStats.recordKeySize(keyLen);
       hostLevelIngestionStats.recordValueSize(valueLen);
+      if (leaderProducedRecordContext != null) {
+        // metrics that are only emitted in leader replica
+        if (isManifestPayload) {
+          hostLevelIngestionStats.recordLeaderLargeValueWrites();
+        } else if (isChunkedDataPayload) {
+          hostLevelIngestionStats.recordLeaderSmallValueWrites();
```

Comment:
Is this a typo? Should recordLeaderChunkDataWrites?

Reply:
Good catch! The condition is wrong. Let me fix it.